server/container_stop.go

```diff
@@ -22,34 +23,31 @@ func (s *Server) StopContainer(ctx context.Context, req *types.StopContainerRequ
 		return nil, status.Errorf(codes.NotFound, "could not find container %q: %v", req.ContainerId, err)
 	}
 
-	sandbox := s.getSandbox(ctx, c.Sandbox())
-	hooks, err := runtimehandlerhooks.GetRuntimeHandlerHooks(ctx, &s.config, sandbox.RuntimeHandler(), sandbox.Annotations())
-	if err != nil {
-		return nil, fmt.Errorf("failed to get runtime handler %q hooks", sandbox.RuntimeHandler())
-	}
-
-	if hooks != nil {
-		if err := hooks.PreStop(ctx, c, sandbox); err != nil {
-			return nil, fmt.Errorf("failed to run pre-stop hook for container %q: %w", c.ID(), err)
-		}
-	}
-
-	if err := s.stopContainer(ctx, c, req.Timeout); err != nil {
+	sb := s.getSandbox(ctx, c.Sandbox())
+	if err := s.stopContainer(ctx, c, sb, req.Timeout); err != nil {
 		return nil, err
 	}
 
-	if err := s.nri.stopContainer(ctx, sandbox, c); err != nil {
-		log.Warnf(ctx, "NRI stop failed for container %q: %v", c.ID(), err)
-	}
-
 	log.Infof(ctx, "Stopped container %s: %s", c.ID(), c.Description())
 	return &types.StopContainerResponse{}, nil
 }
 
 // stopContainer stops a running container with a grace period (i.e., timeout).
-func (s *Server) stopContainer(ctx context.Context, ctr *oci.Container, timeout int64) error {
+func (s *Server) stopContainer(ctx context.Context, ctr *oci.Container, sb *sandbox.Sandbox, timeout int64) error {
```

Comment:
Why don't you add
```go
sb := s.getSandbox(ctx, ctr.Sandbox())
```
instead of adding a new parameter?

Reply:
done thanks!